### PR TITLE
Input/output

### PR DESCRIPTION
**Pull request by AleksandrVoyteshenko**
Made the MainTask and 3 OptionalTask items of the Java input /output section

### CODE REVIEW

File: src/input_and_output/maintask/WorkWithFiles.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            FileOutputStream eleventh2019 = new FileOutputStream("Amon Amarth/2019 - Berserker/11 Wings of Eagles.mp3");
+            FileOutputStream twelfth2019 = new FileOutputStream("Amon Amarth/2019 - Berserker/12 Into the Dark.mp3");
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        writingStructureToFile(album);
+    }
+
+    public static void writingStructureToFile(File album) throws IOException {
```
Comment (nitrogen16) [inline src/input_and_output/maintask/WorkWithFiles.java:122]: Did you try it on any folder?

File: src/input_and_output/optionaltask/first_task/RandomNumbers.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        RandomNumbers.workWithNumbersAndFile(0, 10000);
+        RandomNumbers.readAndSortFile();
+    }
+
+    public static void workWithNumbersAndFile(int min, int max) {
+        int diff = max - min;
+        int[] array = new int[100];
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter("src/input_and_output/optionaltask/first_task/random numbers.txt"));
```
Comment (nitrogen16) [inline src/input_and_output/optionaltask/first_task/RandomNumbers.java:28]: Har coded value. Create a file with path variable

File: src/input_and_output/optionaltask/first_task/RandomNumbers.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void readAndSortFile() {
+        String input;
+        try {
+            ArrayList<Integer> arrayList = new ArrayList<> ();
+            BufferedReader reader = new BufferedReader(new FileReader("src/input_and_output/optionaltask/first_task/random numbers.txt"));
```
Comment (nitrogen16) [inline src/input_and_output/optionaltask/first_task/RandomNumbers.java:43]: Again hard coded path. Avoid this. What if you have 50 methods? 50 times you should repeat it.

File: src/input_and_output/optionaltask/tenth_task/ReadFile.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                    "Я понимаю, что в это мгновенье и в ней и во мне живет одно и то же чувство, что между нами нет никакой разницы.\n" +
+                    "Мы  торжественны: в каждом из нас горит и светится тот же трепетный огонек.\n" +
+                    "Смерть налетит, махнет на него своим холодным широким крылом\n" +
+                    "И конец!\n" +
+                    "Кто потом разберет, какой именно в каждом из нас горел огонек?\n" +
+                    "Нет! Это не животное и не человек меняются взглядами\n" +
+                    "Это две пары одинаковых глаз устремлены друг на друга.\n" +
+                    "И в каждом из этих пар , в животном и в  человеке — одна и та же жизнь жмется пугливо к другой.");
+            writer.close();
+            BufferedReader reader = new BufferedReader(new FileReader("src/input_and_output/optionaltask/tenth_task/text.txt"));
```
Comment (nitrogen16) [inline src/input_and_output/optionaltask/tenth_task/ReadFile.java:46]: Again hard coded value. Avoid this.

File: src/input_and_output/optionaltask/tenth_task/ReadFile.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            while ((line = reader.readLine()) != null) {
+                String[] words = line.split(" ");
+                String temp = words[words.length - 1];
+                words[words.length - 1] = words[0];
+                words[0] = temp;
+                word = Arrays.stream(words).collect(joining(" "));
+                arrayList.add("\n" + word);
+            }
+            reader.close();
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("src/input_and_output/optionaltask/tenth_task/text.txt"));
```
Comment (nitrogen16) [inline src/input_and_output/optionaltask/tenth_task/ReadFile.java:59]: and again

File: src/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            String str = sc.nextLine();
+            boolean indexWord = true;
+            while ((line = reader.readLine()) != null) {
+                indexWord = line.contains(" " + str + " ");
+                if (indexWord == true) {
+                    BufferedWriter writerWord = new BufferedWriter(new FileWriter("src/input_and_output/optionaltask/twelfth_task/word.txt"));
+                    writerWord.write(str);
+                    writerWord.close();
+                } else if (indexWord == false) {
+                    BufferedWriter writerWord = new BufferedWriter(new FileWriter("src/input_and_output/optionaltask/twelfth_task/word.txt"));
```
Comment (nitrogen16) [inline src/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java:40]: 2 instances of BufferedWriter writerWord. Why?

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
src/input_and_output/optionaltask/first_task/random numbers.txt
этот файл генерится как я понимаю
зачем его кидать в ПР?

File: src/input_and_output/maintask/WorkWithFiles.java (hunk 1 above)
Comment (AleksandrVoyteshenko) [reply src/input_and_output/maintask/WorkWithFiles.java:122]: Yes, I did. The method reads the structure of folders and files in the specified directory File album, namely the parent directory of Amon Amarth and writes the structure to the file task.txt

File: src/input_and_output/optionaltask/first_task/RandomNumbers.java (hunk 3 above)
Comment (AleksandrVoyteshenko) [reply src/input_and_output/optionaltask/first_task/RandomNumbers.java:43]: it is necessary to create a file in the directory where the job is located. but it is possible to create a separate file folder for all jobs: src/job_files/

Comment (nitrogen16) [reply src/input_and_output/optionaltask/first_task/RandomNumbers.java:42]: FileReader() retrives a path as a parameter. Therefore, path you use should be as a varible. Better to create a property file.

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
Hard coded paths values

File: src/main/java/input_and_output/optionaltask/first_task/RandomNumbers.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ *      1.     Создать и заполнить файл случайными целыми числами. Отсортировать содержимое файла по возрастанию.
+ */
+
+public class RandomNumbers {
+
+    public static void main(String[] args) {
+        RandomNumbers.workWithNumbersAndFile(0, 10000);
+        RandomNumbers.readAndSortFile();
+    }
```
Comment (nitrogen16) [inline src/main/java/input_and_output/optionaltask/first_task/RandomNumbers.java:21]: [random numbers.txt](https://github.com/AleksandrVoyteshenko/JavaTask/files/5618338/random.numbers.txt)
this is the result....

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                    "Мы  торжественны: в каждом из нас горит и светится тот же трепетный огонек.\n" +
+                    "Смерть налетит, махнет на него своим холодным широким крылом\n" +
+                    "И конец!\n" +
+                    "Кто потом разберет, какой именно в каждом из нас горел огонек?\n" +
+                    "Нет! Это не животное и не человек меняются взглядами\n" +
+                    "Это две пары одинаковых глаз устремлены друг на друга.\n" +
+                    "И в каждом из этих пар , в животном и в  человеке — одна и та же жизнь жмется пугливо к другой.");
+            writer.close();
+            BufferedReader reader = new BufferedReader(new FileReader("src/main/resources/proza.txt"));
+            String line = null;
```
Comment (nitrogen16) [inline src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java:45]: null is redundant

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            String word = null;
```
Comment (nitrogen16) [inline src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java:46]: null is redundant

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+            while ((line = reader.readLine()) != null) {
+                String[] words = line.split(" ");
+                String temp = words[words.length - 1];
+                words[words.length - 1] = words[0];
+                words[0] = temp;
+                word = Arrays.stream(words).collect(joining(" "));
+                arrayList.add("\n" + word);
+            }
+            reader.close();
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("src/main/resources/proza.txt"));
```
Comment (nitrogen16) [inline src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java:57]: "src/main/resources/proza.txt" is used 4 times!!!!!
why?

File: src/main/java/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+            boolean indexWord;
+            while ((line = reader.readLine()) != null) {
+                indexWord = line.contains(" " + str + " ");
+                BufferedWriter writerWord = new BufferedWriter(new FileWriter("src/main/resources/word.txt"));
+                if (indexWord == true) {
+                    writerWord.write(str);
+                    writerWord.close();
+                    break;
+                } else if (indexWord == false) {
+                    writerWord.write("Введенное слово " + str.toUpperCase() + " в тексте не найдено");
```
Comment (nitrogen16) [inline src/main/java/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java:40]: Should we write this word into file if it was not found?

File: src/main/java/input_and_output/maintask/WorkWithFiles.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * Реализовать программу которая будет получать в качестве аргумента командной строки путь к директории, например "D:/movies".
```
Comment (nitrogen16) [inline src/main/java/input_and_output/maintask/WorkWithFiles.java:11]: Реализовать программу которая будет получать в качестве аргумента командной строки путь к директории, например "D:/movies".
Где это реализовано?

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
Fix the comments

File: src/main/java/input_and_output/optionaltask/first_task/RandomNumbers.java (hunk 7 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/optionaltask/first_task/RandomNumbers.java:21]: Yes. What is wrong as a result?

File: src/main/java/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java (hunk 11 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java:40]: I realized that there is no given condition in the task, I replayed it!

File: src/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java (hunk 6 above)
Comment (AleksandrVoyteshenko) [reply src/input_and_output/optionaltask/twelfth_task/WorkWithFileAndText.java]: fixed

File: src/input_and_output/optionaltask/tenth_task/ReadFile.java (hunk 5 above)
Comment (AleksandrVoyteshenko) [reply src/input_and_output/optionaltask/tenth_task/ReadFile.java]: fixed

File: src/input_and_output/optionaltask/tenth_task/ReadFile.java (hunk 4 above)
Comment (AleksandrVoyteshenko) [reply src/input_and_output/optionaltask/tenth_task/ReadFile.java]: fixed

File: src/main/java/input_and_output/maintask/WorkWithFiles.java (hunk 12 above)
Comment (nitrogen16) [reply src/main/java/input_and_output/maintask/WorkWithFiles.java:11]: Was not done!

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
Different folders in path was not implemented.

File: src/main/java/input_and_output/maintask/WorkWithFiles.java (hunk 12 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/maintask/WorkWithFiles.java:10]: fixed

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java (hunk 8 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java]: fixed

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java (hunk 9 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java]: fixed

File: src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java (hunk 10 above)
Comment (AleksandrVoyteshenko) [reply src/main/java/input_and_output/optionaltask/tenth_task/ReadFile.java]: fixed

**Pull request merged**